tensorpack/train/base.py

```diff
@@ -53,6 +55,21 @@ def __init__(self, config):
         self.sess = tf.Session(config=self.config.session_config)
         self.coord = tf.train.Coordinator()
 
+    def apply_tfslim_collections(self, loss_op):
+        # add regularization loss
+        regul_loss = ops.get_collection(ops.GraphKeys.REGULARIZATION_LOSSES)
+        if len(regul_loss) > 0:
+            loss_op += tf.add_n(regul_loss)
+
+        # add non-gradient updates, like batch-norm
+        update_ops = set(ops.get_collection(ops.GraphKeys.UPDATE_OPS))
```

Comment:
@ppwwyyxx Are you worried about the scope? There is a scope-arg in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/g3doc/api_docs/python/functions_and_classes/shard7/tf.get_collection.md